### PR TITLE
feat: Add Balance column on Tokens Page

### LINKED ISSUE #34: Add Balance column on Tokens Page
Feature was requested from MDW. Once implemented we can use balance directly from listing response

Tracking https://github.com/aeternity/ae_mdw/issues/1256

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #34 

## Demo

https://github.com/aeternity/aescan/assets/15363559/83260a38-46f7-45d0-8402-24d2734c7dbf

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] Add amount column to tokens

### CODE REVIEW

File: src/stores/tokens.js [hunk 1]
```diff
@@ -27,15 +36,32 @@ export const useTokensStore = defineStore('tokens', () => {
...
     const { data } = await axios.get(`${useRuntimeConfig().public.MIDDLEWARE_URL}${queryParameters || '/v2/aex9?by=name&direction=forward'}`)
-    allTokens.value = data
+    rawAllTokens.value = data
   }
 
   async function fetchListedTokens() {
     rawListedTokens.value = null
-    const { data } = await axios.get(`${useRuntimeConfig().public.DEX_BACKEND_URL}/tokens`)
-    rawListedTokens.value = data
+    const { data } = await axios.get(`${useRuntimeConfig().public.DEX_BACKEND_URL}/tokens/listed`)
```
Comment (janmichek) [inline src/stores/tokens.js:46]: Additionally, I found out the `/listed` query works, so I could remove it from the frontend (from the adapter)

File: src/stores/tokens.js [hunk 2]
```diff
@@ -27,15 +36,32 @@ export const useTokensStore = defineStore('tokens', () => {
...
+    // for the listed tokens from DEX there need to be amount attached per entry
```
Comment (janmichek) [inline src/stores/tokens.js:47]: not a very nice implementation given the fact we putting more sources together, but the comment explains it.

File: src/utils/hints/tokensHints.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ export const tokensHints = {
   tokenSymbol: 'Non-unique symbol of the token.',
   tokenName: 'Full name of the token.',
   smartContractId: 'Identifier of the Smart Contract that implements the AEX-9 interface.',
+  supplyAmount: 'Initial supply of tokens.',
```
Comment (lukeromanowicz) [inline src/utils/hints/tokensHints.js:5]: What does it mean, do you refer to tokens that have been minted at the moment of contract creation, tokens that have been so far minted or the maximum amount of tokens that can be ever minted?

File: src/components/TokensTable.vue [hunk 4]
```diff
@@ -20,6 +20,12 @@
             {{ tokensHints.smartContractId }}
           </hint-tooltip>
         </th>
+        <th>
+          Amount
```
Comment (lukeromanowicz) [inline src/components/TokensTable.vue:24]: I'd prefer to have a more definitive header name e.g. "Minted Supply", "Mintable Supply" or "Initial Supply" depending on the meaning of this column (refer to my other comment

**Review by lukeromanowicz — APPROVED** [review]
The implementation looks good - nice job :rocket: I commented about some minor details

File: src/components/TokensTable.vue (hunk 4 above)
Comment (janmichek) [reply src/components/TokensTable.vue:24]: Good question, I lost the original intention of this column

**Review by michele-franchi — COMMENTED** [review]
What does exactly this value represents? 
It looks like it's the total supply but with a different formatting.

**Comment by janmichek** [comment]
> What does exactly this value represents? It looks like it's the total supply but with a different formatting.

It represents initial supply from MDW. Is it correct to display it?
image

**Comment by janmichek** [comment]
@michele-franchi It's not very clear what we want to display here.
I gained some answers in here https://github.com/aeternity/ae_mdw/pull/1265#issuecomment-1606916411
Can we make some resolution in here?

**Comment by janmichek** [comment]
Slack convo https://aeternity-workspace.slack.com/archives/C04K21DM61L/p1687764323617789

**Comment by janmichek** [comment]
newer implementation has been done https://github.com/aeternity/aescan/pull/531

**Pull request closed without merging**